### PR TITLE
azurerm_log_profile wait for stable state

### LINKED ISSUE #4559: azurerm_log_profile creation fails due to replication lag
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* `azurerm_log_profile`

### Debug Output

```
Error: Error retrieving Log Profile "foobar": insights.LogProfilesClient#Get: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="ResourceNotFound" Message="Log profile 'foobar' under scope '/subscriptions/xxxx' doesn't exist."

  on ../../../modules/ui_workload/main.tf line 261, in resource "azurerm_monitor_log_profile" "workload_log_profile":
 261: resource "azurerm_monitor_log_profile" "workload_log_profile" {
```

### Expected Behavior

Log profile is created without error messages.

### Actual Behavior

Log profile is created but due to replication lag Terraform sometimes gets a 404 response.

### Steps to Reproduce

Create a basic log profile and run `terraform apply`
Note: This works in most cases so you may have to retry a few times.

### Important Factoids

In most cases this works as expected but sometimes the replication takes a few seconds longer or you are asking an API endpoint which may not have the latest update due to lag.

There is already a `retry` function in place, but this sometimes isn't enough:

1. Create gets 201 response
2. Retry gets 200 response
3. 3rd read gets 404 due to replication lag

How should we approach this? Should we add a `sleep` after the creation? (meh)

Relevant code:
https://github.com/terraform-providers/terraform-provider-azurerm/blob/9608d186084427707a806a4c25e3092f89584d0b/azurerm/resource_arm_monitor_log_profile.go#L142

### PR DESCRIPTION
**Pull request by tiwood**
Our CI/CD pipeline for our infrastructure components fails sometimes due to replication lag while creating `log profiles`.

This PR should fix this by waiting for a stable state.

Fixes #4559

### CODE REVIEW

File: azurerm/resource_arm_monitor_log_profile.go [hunk 1]
```diff
@@ -152,9 +152,19 @@ func resourceArmLogProfileCreateUpdate(d *schema.ResourceData, meta interface{})
 			duration = d.Timeout(schema.TimeoutUpdate)
 		}
 	}
-	// Wait for Log Profile to become available
-	if err := resource.Retry(duration, retryLogProfilesClientGet(ctx, client, name, meta)); err != nil {
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_monitor_log_profile.go]: the linter's complaining about the function `retryLogProfilesClientGet` being unused - could we remove it?

Comment (tiwood) [reply azurerm/resource_arm_monitor_log_profile.go]: Thanks @tombuildsstuff, I've removed the unused func.

**Comment by tiwood** [comment]
@tombuildsstuff, I’ve pushed the requested changes.

**Comment by kikitux** [comment]
@tombuildsstuff can this be added to the changelog for tracking purposes ?

**Pull request merged**